src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,75 +2,97 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 
 import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.Currency;
 import java.util.List;
 
 import static org.junit.Assert.*;
 
 public class ArrayListProductDaoTest
 {
-    private ProductDao productDao;
+    @InjectMocks
+    private final ArrayListProductDao productDao = new ArrayListProductDao();
+
+    private List<Product> testProducts;
+
+    @Mock
+    private Product product1 = new Product();
```

Comment:
You're not creating mock objects here. Mocks are to be created automatically by MockitoJUnitRunner. By explicitly calling constructors here you're creating real objects instead.